packs/vsphere/actions/vm_hw_barebones_create.py

```diff
@@ -0,0 +1,87 @@
+
+        # If No Resource Pool issued the Default one for
+        # the Cluster will be selected.
+        if resourcepool:
+            resource_pool = inventory.get_resource_pool(self.si_content,
+                                                        name=resourcepool)
+        else:
+            resource_pool = cluster.resourcePool
+
+        # Config created that is required for DS selection
+        # Config is BareBones, CPU RAM no more.
+        config = vim.vm.ConfigSpec(name=vm_name,
+                                   memoryMB=(ram_size * 1024),
+                                   numCPUs=cpu_size,
+                                   guestId=guestos,
+                                   version=version,
+                                   cpuHotAddEnabled=True,
+                                   memoryHotAddEnabled=True,
+                                   annotation=description)
+
+        # if Datastore cluster is provided it will find the
+        # recommended Datastore to store VM files
+        if datastore_cluster:
+            podsel = vim.storageDrs.PodSelectionSpec()
+            podsel.storagePod = data_store_cluster
+
+            storageSpec = vim.storageDrs.StoragePlacementSpec(
+                type='create',
+                configSpec=config,
+                resourcePool=resource_pool,
+                podSelectionSpec=podsel,
+                folder=target_folder)
+
+            # Create Instance of Storage Resource Manager -
+            # This is used to identify a Recommended Datastore from a Cluster
+            srm = si_content.storageResourceManager
+            results = srm.RecommendDatastores(storageSpec=storageSpec)
+            rec_ds = results.recommendations[0].action[0]\
+                .relocateSpec.datastore
+            datastore_path = '[' + rec_ds.name + '] ' + vm_name
+        else:
+            # No Datastore Cluster has been offered so using the D
+            if datastore:
+                #data_store = inventory.get_datastore(self.si_content,
+                                                     #name=datastore)
+                datastore_path = '[' + datastore + '] ' + vm_name
+            else:
+                raise Exception('Error No Storage Data Provided')
+
+        #Now Datastore is known the remaining
+        #of VM Config can be setup and added
+        vmx_file = vim.vm.FileInfo(logDirectory=None,
+                                   snapshotDirectory=None,
+                                   suspendDirectory=None,
+                                   vmPathName=datastore_path)
+        config.files = vmx_file
+
+        #Create task to Build actual Machine
+        task = target_folder.CreateVM_Task(config=config, pool=resource_pool)
+        self._wait_for_task(task)
+        if task.info.state != vim.TaskInfo.State.success:
+            raise Exception(task.info.error.msg)
+
+        return {'task_id': task._moId, 'vm_id': task.info.result._moId}
```

Comment:
Since you anyway wait for task returning the taskid likely will not provide much value.